src/ucp/wireup/address.c

```diff
@@ -356,6 +471,81 @@ static void ucp_address_memcheck(ucp_context_h context, void *ptr, size_t size,
     }
 }
 
+static void *ucp_address_pack_byte_extended(void *ptr, size_t value,
+                                            size_t max_value,
+                                            ucp_object_version_t addr_version)
+{
+    if ((addr_version != UCP_OBJECT_VERSION_V1) && (value >= max_value)) {
+        /* Set maximal possible value, indicating that the actual value is in
+         * the next byte. */
+        *ucs_serialize_next(&ptr, uint8_t) = max_value;
+        max_value                          = UINT8_MAX;
+    }
+
+    ucs_assertv_always(value <= max_value, "value=%zu, max_value %zu", value,
+                       max_value);
+
+    *ucs_serialize_next(&ptr, uint8_t) = value;
+
+    return ptr;
+}
+
+static void *
+ucp_address_unpack_byte_extended(const void *ptr, size_t value_mask,
+                                 ucp_object_version_t addr_version,
+                                 uint8_t *value_p)
+{
+    uint8_t value = (*(uint8_t*)ptr) & value_mask;
+
+    if ((addr_version != UCP_OBJECT_VERSION_V1) && (value == value_mask)) {
+        ptr   = UCS_PTR_TYPE_OFFSET(ptr, uint8_t);
+        value = *(uint8_t*)ptr;
+    }
+
+    *value_p = value;
+
+    return UCS_PTR_TYPE_OFFSET(ptr, uint8_t);
```

Comment:
return ptr (assuming prev 2 comments are applied)
and could remove the spaceline before